providers/aws/resources/aws.lr

```diff
@@ -121,6 +121,269 @@ private aws.vpc.flowlog @defaults("id region status") {
   tags map[string]string
 }
 
+// Amazon WAF v2
+private aws.waf {
+  // List of WAF ACLs
+  acls() []aws.waf.acl
+  // List of WAF rules
+  ruleGroups() []aws.waf.rulegroup
+  // List of WAF IP sets
+  ipSets() []aws.waf.ipset
+}
+
+// Amazon WAF v2 ACL
+private aws.waf.acl @defaults("name") {
+  // ARN of the ACL
+  arn string
+  // ID of the ACL
+  id string
+  // Name of the ACL
+  name string
+  // Description of the ACL
+  description string
+  // Whether the ACL is managed by Firewall Manager
+  managedByFirewallManager bool
+  // List of WAF rules
+  rules() []aws.waf.rule
+}
+
+// Amazon WAF rule
+private aws.waf.rule @defaults("name") {
+  // Name of the rule
+  name string
+  // Priority
+  priority int
+  // Part of the rule that tells WAF how to inspect a web request
+  statement aws.waf.rule.statement
+  // Part of the rule that tells WAF what to do with a web request when it matches the criteria defined in the rule
+  action aws.waf.rule.action
+}
+
+private aws.waf.rule.action {
+  allow aws.waf.rule.action.allow
+  block aws.waf.rule.action.block
+  count aws.waf.rule.action.count
+  captcha aws.waf.rule.action.captcha
+}
+
+private aws.waf.rule.action.allow { }
+private aws.waf.rule.action.block { }
+private aws.waf.rule.action.count { }
+private aws.waf.rule.action.captcha { }
+
+private aws.waf.rule.statement {
+  // random uuid, to make the statement unique, these don't exist in AWS
+  id string
+  sqliMatchStatement aws.waf.rule.statement.sqlimatchstatement
```

Comment:
I have been mimicking what the aws api is giving me, how else would you do it? I could add an additional field `kind` that contains the kind of statement we are dealing with.

Reply:
So I could try to flatten the whole thing so that we have something like

```
rule.statement { 
  kind: sqliMatchStatement
  sensitivityLevel: low
  ... whatever other fields we need for a sqliMatchStatement
}

Reply:
What do you think about the statement like this

```
aws.waf.acls.first.rules[1].statement: {
  size: null
  searchString: "zzzzz"
  statements: null
  key: null
  fallbackBehavior: null
  VendorName: null
  ruleName: "marius-test-two"
  Name: null
  kind: "ByteMatchStatement"
  headerName: null
  statement: null
  fieldToMatch: aws.waf.rule.fieldtomatch id = marius-test-two
  countryCodes: null
  regexString: null
  comparisonOperator: null
  position: null
  scope: null
  arn: null
```

This puts all the fields directly on the `statement`, I added the `kind` field to tell what kind of statement it is - all fields that are not part of a `ByteMatchStatement` in this case are null - would that be better?

Reply:
For the record, I ended up not doing the above, we still have all the subtypes of statements. Even if we add all fields to the statement, the users still has to first check what kind of statement he is dealing with (which he can now do with the field `kind` ).
> This makes it difficult to check, since you as a user do not really know which check type is used
I don't thing that there is any way around this, you have to first know which check type is used. If you don't know which check type is used you can't know what fields you would be looking for.